staging/src/k8s.io/apimachinery/pkg/util/proxy/upgradeaware.go

```diff
@@ -383,6 +384,13 @@ func (h *UpgradeAwareHandler) tryUpgrade(w http.ResponseWriter, req *http.Reques
 	}
 	defer requestHijackedConn.Close()
 
+	// Occurs immediately after a successful Hijack call to bypass cases where the parsed
+	// backend status is not necessarily the same status we decide to write in the response to
+	// the client.
+	for _, recorder := range statusRecordersFrom(req.Context()) {
+		recorder.RecordStatus(backendHTTPResponse.StatusCode)
```

Comment:
Since `StatusCode` is unbounded here, we may end up with a high cardinality label when generating the counter.

https://github.com/kubernetes/kubernetes/blob/235d095517da0d285a7fe6c2f12b01370ca7a710/staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go#L530